processor/src/host/debug.rs

```diff
@@ -71,6 +74,44 @@ impl Printer {
         }
     }
 
+    /// Prints the number of advice stack items specified by `n` if it is provided, otherwise print
```

Comment:
Seems a bit odd to only print the top item if no size is specified. I'd expect that specifying no "limit"/size would print the entire stack, not just the top (for which you'd presumably pass `Some(1)` instead).